### PR TITLE
DTFS2- 5444 - ACCESSIBILITY -  non descriptive links

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made: 

* Changed links to be more descriptive
* For bond and loan, instead of not entered, changed to bond/loan reference number not entered
* Added internal bank reference name to clone and abandon links text
* For delete, changed to delete loan/bond
* Added deal name to edit link

### CODE REVIEW

File: portal/templates/contract/components/abandon-deal-link.njk [hunk 1]
```diff
@@ -9,8 +9,9 @@
           data-cy="AbandonLink"
           href="/contract/{{ params.deal._id }}/delete"
           class="govuk-link govuk-!-font-size-14 govuk-!-margin-left-2"
-          aria-label="Abandon Deal"
-        >Abandon</a>
+          title="Abandon Deal {{ params.deal.bankInternalRefName }}"
+          aria-label="Abandon Deal {{ params.deal.bankInternalRefName }}"
+        >Abandon {{ params.deal.bankInternalRefName }}</a>
```
Comment (IgnatG) [inline portal/templates/contract/components/abandon-deal-link.njk:14]: how does it render on the page if the bankInternalRefName is very long?

File: portal/templates/contract/components/clone-deal-link.njk [hunk 2]
```diff
@@ -8,8 +8,9 @@
         href="/contract/{{ params.deal._id }}/clone/before-you-start"
         class="govuk-link govuk-!-font-size-14"
         data-cy="clone-deal-link"
-        aria-label="Clone Deal"
-      >Clone</a>
+        title="Clone Deal {{ params.deal.bankInternalRefName }}"
+        aria-label="Clone Deal {{ params.deal.bankInternalRefName }}"
+      >Clone {{ params.deal.bankInternalRefName }}</a>
```
Comment (IgnatG) [inline portal/templates/contract/components/clone-deal-link.njk:13]: same as above

Comment (IgnatG) [reply portal/templates/contract/components/clone-deal-link.njk:13]: also, is this applicable to both BSS and GEF deals?

File: portal/templates/contract/components/abandon-deal-link.njk (hunk 1 above)
Comment (IgnatG) [reply portal/templates/contract/components/abandon-deal-link.njk:14]: and is this applicable to both GEF and BSS deals?

Comment (Zainzzkk) [reply portal/templates/contract/components/abandon-deal-link.njk:14]: sent what it looks like on slack.  And currently only on BSS.  Not got the tickets for GEF analysis yet

File: portal/templates/contract/components/clone-deal-link.njk (hunk 2 above)
Comment (Zainzzkk) [reply portal/templates/contract/components/clone-deal-link.njk:13]: sent what it looks like on slack.  And currently only on BSS.  Not got the tickets for GEF analysis yet

File: portal/templates/contract/components/abandon-deal-link.njk (hunk 1 above)
Comment (Zainzzkk) [reply portal/templates/contract/components/abandon-deal-link.njk]: done

File: portal/templates/contract/components/issue-or-delete-facility-link.njk [hunk 3]
```diff
@@ -27,8 +27,11 @@
       <a href="/contract/{{ dealId }}/submission-details#{{ facilityName }}-{{ facility._id }}" class="govuk-link" data-cy="{{ facilityName }}-issue-facility-{{ facility._id }}">Facility issued</a>
 
     {% elseif (userIsMaker and dealIsEditable) and (dealStatus === "Draft" or dealStatus === "Further Maker's input required") and not dealSubmissionDate %}
-      <a href="/contract/{{ dealId }}/{{ facilityName }}/{{ facility._id}}/delete" class="govuk-link" data-cy="{{ facilityName }}-delete-{{ facility._id }}" aria-label="Delete {{ facilityName }} {{ facilityTableIndex }}">Delete</a>
-
+      {% if facility.name %}
+        <a href="/contract/{{ dealId }}/{{ facilityName }}/{{ facility._id}}/delete" class="govuk-link" data-cy="{{ facilityName }}-delete-{{ facility._id }}" aria-label="Delete {{ facilityName }} {{ facility.name }}">Delete {{ facility.name }}</a>
```
Comment (IgnatG) [inline portal/templates/contract/components/issue-or-delete-facility-link.njk:31]: what is the difference between `facilityName` and `facility.name` ?

Comment (Zainzzkk) [reply portal/templates/contract/components/issue-or-delete-facility-link.njk:31]: facityName is the type, like loan or bond, but facility.name is the name set to the facility (not always set)

File: portal/templates/contract/components/issue-or-delete-facility-link.njk [hunk 4]
```diff
@@ -27,8 +27,11 @@
...
+      {% else %}
+        <a href="/contract/{{ dealId }}/{{ facilityName }}/{{ facility._id}}/delete" class="govuk-link" data-cy="{{ facilityName }}-delete-{{ facility._id }}" aria-label="Delete {{ facilityName }} {{ facilityTableIndex }}">Delete {{ facilityName }}</a>
```
[the remark this answers is not in the document]
Comment (Zainzzkk) [reply portal/templates/contract/components/issue-or-delete-facility-link.njk:33]: say if it is delete loan 0123 - if name is set, then it says delete loan 123.  (if I also add the index, it says delete loan 123 1)
but if name not set, it says delete loan 1 (which is what it was before)

File: trade-finance-manager-api/src/graphql/reducers/mappings/gef-facilities/mapGefFacilityDates.api-test.js [hunk 5]
```diff
@@ -72,7 +72,7 @@ describe('mapGefFacilityDates', () => {
     expect(result.tenor).toEqual(expected);
   });
 
-  it('should return object with undefined coverEndDate if coverEndDate is null', () => {
+  it('could just say should not map coverEndDate when date is null', () => {
```
Comment (IgnatG) [inline trade-finance-manager-api/src/graphql/reducers/mappings/gef-facilities/mapGefFacilityDates.api-test.js:75]: I'm not sure I understand the title of this test?

Comment (Zainzzkk) [reply trade-finance-manager-api/src/graphql/reducers/mappings/gef-facilities/mapGefFacilityDates.api-test.js:75]: I dunno how I managed to do that 😂

Comment (Zainzzkk) [reply trade-finance-manager-api/src/graphql/reducers/mappings/gef-facilities/mapGefFacilityDates.api-test.js]: done

**Review by IgnatG — APPROVED** [review]

**Pull request merged**